src/data/services/LmsApiService.js

```diff
@@ -70,11 +70,13 @@ class LmsApiService {
   }
 
   static getGradeExportCsvUrl(courseId, options = {}) {
-    const trackQueryParam = options.track ? [`track=${options.track}`] : [];
-    const cohortQueryParam = options.cohort ? [`cohort=${options.cohort}`] : [];
-    const queryParams = [...trackQueryParam, ...cohortQueryParam].join('&');
-    const downloadUrl = `${LmsApiService.baseUrl}/api/bulk_grades/course/${courseId}/?${queryParams}`;
-    return downloadUrl;
+    const queryParams = ['track', 'cohort', 'assignment', 'assignmentType']
+      .filter(opt => Object.prototype.hasOwnProperty.call(options, opt) &&
```

Comment:
What does the `Object.prototype.hasOwnProperty.call(options, opt)` call tell you? I understand that it checks whether `opt` is a property of the `options` object. But why is that piece of information important? I would think it would be enough to check whether the `opt` key exists in `options`, which it seems like you're doing in the second condition in the expression. Could you explain?

Reply:
Is there a difference between `Object.prototype.hasOwnProperty.call(options, opt)` and `options.prototype.hasOwnProperty(opt)`? Or are they equivalent?

Reply:
To your second question, they are not: https://eslint.org/docs/rules/no-prototype-builtins

Generally `options.prototype.hasOwnProperty(opt)` wouldn't do what you may be thinking it'd do. That'd check if the prototype object has the property, not whether `options` has the property.

It is perfectly reasonable to contrast between what I have here and `options.hasOwnProperty(opt)`, and the difference there is that this function could be called with an `options` object of `{ hasOwnProperty: () => true }`, which would end up being called before we got to the prototype implementation. This isn't actually a case where there's any risk of that, but as usual eslint cannot be reasoned with per se.

To your first question, I think I wrote the first condition first and added the other conditions later after I'd plugged this function into its call sites, finding them not covering the cases I needed. I'll remove this condition because it is redundant in context. (There are inputs for which this will behave differently without this condition, but they are highly contrived and involve mucking with prototypes of these objects.)